data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -171,6 +176,10 @@ public static List<DataFile> listPartition(
     }
   }
 
+  public static boolean isFileReadingParallelized() {
```

Comment:
I don't think it's a good idea to introduce this, since this reflects whatever was set after calling `listPartition(...)` and depending on the order of calls, it could return something unexpected

Reply:
@nastra do you have an idea to test whether parallelism setting is working?

Reply:
I haven't checked the details on how to properly test this, but introducing a static variable isn't the right way